src/ape/managers/chain.py

```diff
@@ -1,18 +1,23 @@
 import time
-from typing import Callable, ClassVar, Dict, Iterator, List, Optional, Tuple, Union
+from typing import Callable, ClassVar, Dict, Iterator, List, Optional, Tuple, Union, cast
+
+import pandas as pd
 
 from ape.api import AddressAPI, BlockAPI, ProviderAPI, ReceiptAPI
+from ape.api.query import BlockQuery, QueryAPI
 from ape.exceptions import ChainError, ProviderNotConnectedError, UnknownSnapshotError
 from ape.logging import logger
 from ape.managers.converters import ConversionManager
 from ape.managers.networks import NetworkManager
+from ape.managers.query import QueryManager
 from ape.types import AddressType, BlockID, SnapshotID
 from ape.utils import cached_property, injected_before_use
 
 
 class _ConnectedChain:
     _networks: ClassVar[NetworkManager] = injected_before_use()  # type: ignore
     _converters: ClassVar[ConversionManager] = injected_before_use()  # type: ignore
+    _query_manager: ClassVar[QueryManager] = cast(QueryManager, injected_before_use())
```

Comment:
when do we use `cast()` versus `type: ignore`?

Reply:
This will all be changed in the pydantic refactor, so I was just adding it the newer way but ultimately it won't be here. All the managers will be set in the `ManagerAccessBase` class.